malloc/malloc.c

```diff
@@ -40,7 +67,31 @@ void mem_copy(void *dest, const void *src, int n)
 */
 void *resize_memory(void *ptr, int old_size, int new_size)
 {
+    // making new memory allocation to transfer data to
+    char *new_malloc = malloc(new_size);
+    char *cast_ptr = (char *)ptr;
+    int i;
+
+    if (old_size < new_size)
+    {
+        // looping to old_size value in old malloc
+        for (i = 0; i < old_size; i++)
+        {
+            // mapping values in old malloc to new malloc
+            *(new_malloc + i) = *(cast_ptr + i);
+        }
+    }
+    else if (old_size >= new_size)
+    {
+        // looping to new_size value in old malloc
+        for (i = 0; i < new_size; i++)
+        {
+            // mapping values in old malloc to new malloc
+            *(new_malloc + i) = *(cast_ptr + i);
+        }
+    }
 
```

Comment:
once you resize ptr, ptr becomes obsolete, so you should free up its memory space before this function ends